src/naming.test.js

```diff
@@ -40,15 +40,33 @@ describe('#naming', function () {
     let naming = null;
     beforeEach(() => naming = new Naming());
 
-    it('should interpolate alarm name', () => {
-      const template = '$[functionName]-$[functionId]-$[metricName]-$[metricId]';
+    [true, false, undefined].forEach(stackNamePrefix => {
+      const prefixDescription = stackNamePrefix === false ? '' : 'and prefix with stack name';
+      it(`should interpolate alarm name ${prefixDescription}`, () => {
+        const template = '$[functionName]-$[functionId]-$[metricName]-$[metricId]';
+        const functionName = 'function';
+        const functionLogicalId = 'functionId';
+        const metricName = 'metric';
+        const metricId = 'metricId';
+        const stackName = 'fooservice-dev';
+        const expectedPrefix = stackNamePrefix === false ? '' : `${stackName}-`;
+
+        const expected = `${expectedPrefix}${functionName}-${functionLogicalId}-${metricName}-${metricId}`;
+        const actual = naming.getAlarmName({ stackNamePrefix, template, functionName, functionLogicalId, metricName, metricId, stackName });
+
+        expect(actual).toEqual(expected);
+      });
+    });
+
+    it('should not prefix when $[stackName] is in the template', () => {
+      const template = '$[functionName]-$[stackName]-$[functionId]-$[metricName]-$[metricId]';
```

Comment:
nit: you can reduce this to `$[functionName]-$[stackName]`, since everything else is not relevant to the test

Reply:
done